primitives/src/vid/advz.rs

```diff
@@ -483,6 +485,10 @@ where
     fn get_payload_byte_len(common: &Self::Common) -> usize {
         common.bytes_len
     }
+
+    fn get_num_storage_nodes(common: &Self::Common) -> usize {
+        common.num_storage_nodes
```

Comment:
Don't we have to verify somewhere that this is consistent with the actual matrix encoding of the data? I would expect to see changes somewhere in `verify_share` or something. Same question also for `bytes_len`

Reply:
Good questions.

# Number of storage nodes

We can check this against the VID scheme params in any method that takes both `self` and `common`. Currently that's `verify_share` and `recover_payload`. I'll add this.

I just noticed that this quantity should be hashed into the commitment like we already do for `bytes_len`:
https://github.com/EspressoSystems/jellyfish/blob/bf1a6dfb63e2ce9b409a682d2793b836f1659097/primitives/src/vid/advz.rs#L543-L547
I'll add this.

# Payload byte length

The exact byte length would be lost in `disperse` if we did not save it in `Common`. The matrix gives you only an upper bound on the byte length, as we pad the payload with zero-bytes to fill the matrix. It is used within jellyfish only in `recover_payload` so that we can truncate the recovered payload to the correct length.

I don't see anything we could do here to enforce a declaration by the disperser of the "correct" byte length.

Reply:
Ah I see. The correct byte length is just a convention, so we don't have to enforce it, we just need agreement. Maybe we should enforce that it is plausible, though (ie less than or equal to the padded byte length)

Reply:
Sorry, this is out of scope for this PR. I'll approve and you can decide whether to add this stuff now or in a separate PR

Reply:
> Ah I see. The correct byte length is just a convention, so we don't have to enforce it, we just need agreement. Maybe we should enforce that it is plausible, though (ie less than or equal to the padded byte length)

The only place where we could execute this enforcement is in `verify_share` by rejecting a share if the claimed payload byte len is larger than what's possible given [num polynomials] and [polynomial degree]. But is this really worth it? An alternative is to simply *define* the payload byte length as the *minimum* of (i) what's possible given [num polynomials] and [polynomial degree], and (ii) the declared byte len in `Common`. ie. If it's too big then `truncate` does nothing. Complexity is reduced.

Reply:
Punted to #473 . I'm happy as-is for now, at least.